internal/kube/metrics/client.go

```diff
@@ -0,0 +1,77 @@
+package metrics
+
+import (
+	"context"
+	"net/url"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/client-go/tools/metrics"
+)
+
+// MustRegisterClientGoMetrics registers a set of metrics exposed from the
+// k8s.io/client-go/tools/metrics package with the prometheus registry.
+func MustRegisterClientGoMetrics(registry *prometheus.Registry) {
+	httpMetrics := &clientGoHttpMetrics{
+		latency: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Namespace: "skupper",
+			Subsystem: "kubernetes_client",
+			Name:      "http_request_duration_seconds",
+			Help:      "Latency of kubernetes client requests in seconds by endpoint.",
+		}, []string{"method", "endpoint"}),
+		results: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: "skupper",
+			Subsystem: "kubernetes_client",
+			Name:      "http_requests_total",
+			Help:      "Total number of kubernetes client requests by status code.",
+		}, []string{"status_code"}),
```

Comment:
Minor suggestion just to add the method to this and to the http_retries_total as well.